engine/src/main/java/io/camunda/zeebe/streamprocessor/ErrorHandlingContextImpl.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.streamprocessor;
+
+import io.camunda.zeebe.engine.api.ErrorHandlingContext;
+import io.camunda.zeebe.engine.api.ProcessingResultBuilder;
+
+final class ErrorHandlingContextImpl implements ErrorHandlingContext {
```

Comment:
:wrench: If we don't plan on adding much logic to these (I assume) or making them mutable (I wish), these could be `record` types.

Reply:
My plan is to remove this interface and implementation soon.